js/julia.js

```diff
@@ -169,6 +169,40 @@ CodeMirror.defineMode("julia2", function(config, parserConfig) {
     var leaving_expr = state.leaving_expr;
     state.leaving_expr = false;
 
+    // Number Literals
+    if (stream.match(/^[+-]?[0-9\.]/, false)) {
+
+      var float_literal = false;
+
+      if (stream.match(/^[\d_]+\.[\d_]*([Eef][+-]?\d+)?(im)?/)) {
+        float_literal = true;
+      } else if (stream.match(/^[\d_]*\.[\d_]+([Eef][+-]?\d+)?(im)?/)) {
+        float_literal = true;
+      }
+
+      if (float_literal) {
+        finalise_leavingexpr(state, stream);
+        return 'number';
+      }
+
+      var int_literal = false;
+
+      if (stream.match(/^0x[0-9a-f_]+(\.?[0-9a-f_]*[p][+-]?\d+)?(im)?(?![\d\.])/i)) {
+        int_literal = true;
+      } else if (stream.match(/^0b[01_]+(im)?(?![\d\.])/)) {
+        int_literal = true;
+      } else if (stream.match(/^0o[0-7_]+(im)?(?![\d\.])/)) {
+        int_literal = true;
+      } else if (stream.match(/^[\d_]+([Eef][\+\-]?\d+)?(im)?(?![\d\.])/)) {
+        int_literal = true;
+      }
+
+      if (int_literal || stream.match(/^0(?!\d)(im)?/)) {
+        finalise_leavingexpr(state, stream);
+        return 'number';
+      }
+    }
+
     // If a comma or dot, don't remove last keyword
     // This is so that `using A, B` works
     if (stream.match(/\.|,/))
```

Comment:
I had to move the number's section above this otherwise floating point numbers weren't highlighted correctly. I've not noticed any problems by doing that, but I'm not completely sure whether the ordering is important here?